cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -2163,7 +2163,7 @@
 ;; even not generate a zero-extended move in this case.
 (rule 1 (extend_to_gpr src @ (value_type $I32) $I64 (ExtendKind.Zero))
         (if-let $true (value32_zeros_upper32 src))
-        src)
+        (add_range_fact src 64 0 0xffff_ffff))
```

Comment:
I think this would be a little bit more general and less ad-hoc or at least more aesthetically pleasing to me if we separated fact attachment from fact construction:

```suggestion
        (let ((fact Fact (make_range_fact 64 0 0xffff_ffff)))
          (with_fact src fact)))
```

But feel free to take or leave this suggestion.

Reply:
For now I'm going to keep this just for expediency / YAGNI but let's refactor as needed when we come to other sorts of fact propagation!